lib/Models/Catalog/CatalogItems/CzmlCatalogItem.ts

```diff
@@ -89,34 +89,35 @@ export default class CzmlCatalogItem
     return isDefined(this._czmlFile);
   }
 
-  @action
   protected forceLoadMapItems(): Promise<void> {
     const attribution = this.attribution;
-    return new Promise<string | readonly JsonObject[]>(resolve => {
-      if (isDefined(this.czmlData)) {
-        resolve(toJS(this.czmlData));
-      } else if (isDefined(this.czmlString)) {
-        resolve(JSON.parse(this.czmlString));
-      } else if (isDefined(this._czmlFile)) {
-        resolve(readJson(this._czmlFile));
-      } else if (isDefined(this.url)) {
-        resolve(proxyCatalogItemUrl(this, this.url, this.cacheDuration));
-      } else {
-        throw new TerriaError({
-          sender: this,
-          title: i18next.t("models.czml.unableToLoadItemTitle"),
-          message: i18next.t("models.czml.unableToLoadItemMessage")
-        });
-      }
-    })
-      .then(czmlLoadInput =>
-        CzmlDataSource.load(czmlLoadInput, {
-          credit: attribution
-        })
-      )
+    let loadableData: any = undefined;
+    if (isDefined(this.czmlData)) {
+      loadableData = toJS(this.czmlData);
+    } else if (isDefined(this.czmlString)) {
+      loadableData = JSON.parse(this.czmlString);
+    } else if (isDefined(this._czmlFile)) {
+      loadableData = readJson(this._czmlFile);
+    } else if (isDefined(this.url)) {
+      loadableData = proxyCatalogItemUrl(this, this.url, this.cacheDuration);
+    }
+
+    if (loadableData === undefined) {
+      throw new TerriaError({
+        sender: this,
+        title: i18next.t("models.czml.unableToLoadItemTitle"),
+        message: i18next.t("models.czml.unableToLoadItemMessage")
+      });
+    }
+
+    return makeRealPromise<CzmlDataSource>(
+      CzmlDataSource.load(loadableData, {
+        credit: attribution
+      })
+    )
```

Comment:
Moved the checks to top level so that `mobx` correctly registers the dependency on the values.